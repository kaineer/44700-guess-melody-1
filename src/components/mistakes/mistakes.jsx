import React from 'react';
import {connect} from 'react-redux';
import {number} from 'prop-types';

export const Mistakes = ({mistakes}) => (
  <div className="game__mistakes">
    {(new Array(mistakes)).fill(1).map((value, i) => (<div key={`mistake-${i}-${value}`} className="wrong"></div>))}
  </div>
);

Mistakes.propTypes = {
  mistakes: number.isRequired
};

const mapStateToProps = ({mistakes}, ownProps) => Object.assign({}, ownProps, {
  mistakes
});

export default connect(mapStateToProps, null)(Mistakes);
