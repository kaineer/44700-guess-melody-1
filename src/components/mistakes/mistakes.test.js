// src/components/mistakes.test.js

import React from 'react';
import {Mistakes} from './mistakes';
import {create} from 'react-test-renderer';

describe(`<Mistakes />`, () => {
  it(`should render correctly for no mistakes`, () => {
    const tree = create(
      <Mistakes
        mistakes={0}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render correctly for three mistakes`, () => {
    const tree = create(
      <Mistakes
        mistakes={3}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
