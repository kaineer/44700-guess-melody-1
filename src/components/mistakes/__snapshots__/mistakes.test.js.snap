// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mistakes /> should render correctly for no mistakes 1`] = `
<div
  className="game__mistakes"
/>
`;

exports[`<Mistakes /> should render correctly for three mistakes 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;
